src/components/modebar/modebar.js

```diff
@@ -173,31 +180,42 @@ proto.createButton = function(config) {
  * @Param {object} thisIcon
  * @Param {number} thisIcon.width
  * @Param {string} thisIcon.path
+ * @Param {string} thisIcon.color
  * @Return {HTMLelement}
  */
 proto.createIcon = function(thisIcon) {
     var iconHeight = isNumeric(thisIcon.height) ?
             Number(thisIcon.height) :
             thisIcon.ascent - thisIcon.descent,
         svgNS = 'http://www.w3.org/2000/svg',
-        icon = document.createElementNS(svgNS, 'svg'),
-        path = document.createElementNS(svgNS, 'path');
+        icon = document.createElementNS(svgNS, 'svg');
 
     icon.setAttribute('height', '1em');
     icon.setAttribute('width', (thisIcon.width / iconHeight) + 'em');
     icon.setAttribute('viewBox', [0, 0, thisIcon.width, iconHeight].join(' '));
 
-    path.setAttribute('d', thisIcon.path);
+    if(thisIcon.path) {
+        var path = document.createElementNS(svgNS, 'path');
+        path.setAttribute('d', thisIcon.path);
 
-    if(thisIcon.transform) {
-        path.setAttribute('transform', thisIcon.transform);
-    }
-    else if(thisIcon.ascent !== undefined) {
-        // Legacy icon transform calculation
-        path.setAttribute('transform', 'matrix(1 0 0 -1 0 ' + thisIcon.ascent + ')');
+        if(thisIcon.transform) {
+            path.setAttribute('transform', thisIcon.transform);
+        }
+        else if(thisIcon.ascent !== undefined) {
+            // Legacy icon transform calculation
+            path.setAttribute('transform', 'matrix(1 0 0 -1 0 ' + thisIcon.ascent + ')');
+        }
+
+        if(thisIcon.color) {
+            path.setAttribute('fill', thisIcon.color);
+        }
+
+        icon.appendChild(path);
     }
 
-    icon.appendChild(path);
+    if(thisIcon.svg) {
+        icon.innerHTML = thisIcon.svg;
```

Comment:
`test-syntax` fails because or `innerHTML` :(

Reply:
This should be fine. `innerHTML` only causes problems on SVG nodes. 

Looks like the concerns written down in 

https://github.com/plotly/plotly.js/blob/9772ef6dd5ec5417ab50dd378aaa536a8c845336/tasks/test_syntax.js#L106-L109

have become reality.

Reply:
`thisIcon.svg` isn't a SVG document but rather its content. `icon` is the SVG document so IE doesn't like this. Let me fix this :)

Reply:
Fixed in commit e870e34

Reply:
I got rid of `innerHTML` altogether and now use `DOMParser` in commit 61d1c7b !